sqlx-core/src/any/transaction.rs

```diff
@@ -10,27 +10,40 @@ pub struct AnyTransactionManager;
 
 impl TransactionManager for AnyTransactionManager {
     type Database = Any;
+    type Options = ();
```

Comment:
Actually, I think this could be a struct containing a `TransactionOptions` for each database, and then the implementation picks the corresponding one:

```rust
#[derive(Default)]
pub struct AnyTransactionOptions {
    #[cfg(feature = "postgres")]
    postgres: PgTransactionOptions

    #[cfg(feature = "mysql")]
    mysql: MySqlTransactionOptions,

    // etc.
}
```

Reply:
This could then have setters to replace each type wholesale:

```rust
impl AnyTransactionOptions {
    #[cfg(feature = "postgres")]
    pub fn postgres(self, postgres: PgTransactionOptions) -> Self {
        Self { postgres, ..self }
    }

    // etc.
}
```

Or even have setters for "best effort" options:

```rust
impl AnyTransactionOptions {
    pub fn isolation_level_if_supported(self, ilvl: IsolationLevel) -> Self {
        #[cfg(feature = "postgres")]
        self.postgres = self.postgres.isolation_level(ilvl);

        #[cfg(feature = "mysql")]
        self.mysql = self.mysql.isolation_level(ilvl);

        self
    }
}
```

(Since `PgIsolationLevel` and `MySqlIsolationLevel` are identical, they don't need to be separate enums.)